esmvalcore/_data_finder.py

```diff
@@ -56,6 +69,9 @@ def get_start_end_year(filename):
 
     # As final resort, try to get the dates from the file contents
     if start_year is None or end_year is None:
+        message = f"Must load file {filename} for daterange " + \
+                  f"(because filename produced {daterange})"
+        logger.debug(message)
```

Comment:
```suggestion
        logger.debug(f"Must load file {filename} for daterange "
                     f"(because filename produced {daterange})")
```

Reply:
Done